query_frontend/src/provider.rs

```diff
@@ -138,6 +153,34 @@ impl<'a> MetaProvider for CatalogMetaProvider<'a> {
     fn aggregate_udf(&self, name: &str) -> Result<Option<AggregateUdf>> {
         self.function_registry.find_udaf(name).context(FindUdf)
     }
+
+    // TODO: after supporting not only default catalog and schema, we should
+    // refactor the tables collecting procedure.
+    fn all_tables(&self) -> Result<Vec<TableRef>> {
+        let catalog = match self
+            .manager
+            .catalog_by_name(self.default_catalog)
+            .with_context(|| FindCatalog {
+                name: self.default_catalog,
+            })? {
+            Some(catalog) => catalog,
+            None => return Ok(Vec::default()),
```

Comment:
Why not throw error when no catalog isn't found?

Reply:
I think it again, better to return error really.